pumps/mongo_aggregate.go

```diff
@@ -261,12 +288,30 @@ func (m *MongoAggregatePump) WriteData(ctx context.Context, data []interface{})
 		m.WriteData(ctx, data)
 	} else {
 		// calculate aggregates
-		analyticsPerOrg := analytics.AggregateData(data, m.dbConf.TrackAllPaths, m.dbConf.IgnoreTagPrefixList, m.dbConf.StoreAnalyticsPerMinute, true)
-
+		analyticsPerOrg := analytics.AggregateData(data, m.dbConf.TrackAllPaths, m.dbConf.IgnoreTagPrefixList, m.dbConf.MongoURL, m.dbConf.AggregationTime, true)
 		// put aggregated data into MongoDB
 		for orgID, filteredData := range analyticsPerOrg {
 			err := m.DoAggregatedWriting(ctx, orgID, filteredData)
 			if err != nil {
+				// checking if the error is related to the document size and AggregateSelfHealing is enabled
+				if strings.Contains(err.Error(), "Size must be between 0 and 16793600(16MB)") && m.dbConf.EnableAggregateSelfHealing {
```

Comment:
Would be the same error for the any supported DB that the customer use? Am wondering if this would be the same for cosmo,mongo, documentDB or should we isolate this error into its own function ?

Reply:
good point @sredxny. @mativm02 did you have a look at CosmoDB/DocDB?

Reply:
Oops, I checked with CosmosDB and it throws a different error! Will find a way to handle this situation.